packages/common/src/search/_internal/hubSearchItemsHelpers/formatOgcItemsResponse.ts

```diff
@@ -1,16 +1,56 @@
 import { IQuery } from "../../types/IHubCatalog";
 import { IHubSearchOptions } from "../../types/IHubSearchOptions";
 import { IHubSearchResponse } from "../../types/IHubSearchResponse";
-import { IHubSearchResult } from "../../types/IHubSearchResult";
 import { getNextOgcCallback } from "./getNextOgcCallback";
 import { IOgcItemsResponse } from "./interfaces";
 import { ogcItemToSearchResult } from "./ogcItemToSearchResult";
+import { ogcItemToDiscussionPostResult } from "./ogcItemToDiscussionPostResult";
+import { HubOgcSearchResultType } from "../../types/HubOgcSearchResultType";
 
 export async function formatOgcItemsResponse(
   response: IOgcItemsResponse,
   originalQuery: IQuery,
   originalOptions: IHubSearchOptions
-): Promise<IHubSearchResponse<IHubSearchResult>> {
+): Promise<IHubSearchResponse<HubOgcSearchResultType>> {
+  if (originalQuery.targetEntity === "discussionPost") {
+    return formatDiscussionPoastTargetEntityResponse(
+      response,
+      originalQuery,
+      originalOptions
+    );
+  }
+
+  return formatItemTargetEntityResponse(
+    response,
+    originalQuery,
+    originalOptions
+  );
+}
+
+async function formatDiscussionPoastTargetEntityResponse(
```

Comment:
Minor - funciton name `Post` vs `Poast`

```suggestion
async function formatDiscussionPostTargetEntityResponse(
```

Reply:
great catch! This has been fixed.